File: csharp/math/Pascal.cs
// Print 10 rows of Pascal's triangle, each row rewritten in place from its right hand end.
// Run: dotnet run Pascal.cs (file-based app, .NET 10 or later), or as Program.cs in a console project

const int rows = 10;

var row = new long[rows + 1];
row[0] = 1;

for (var length = 1; length <= rows; length++)
{
    Console.WriteLine(string.Join(" ", row.Take(length)));

    for (var i = length; i > 0; i--)
    {
        row[i] += row[i - 1];
    }
}

File: go/math/pascal.go
// Print 10 rows of Pascal's triangle, each row grown by append and accumulated from the back.
// Run: go run pascal.go

package main

import (
	"fmt"
	"strconv"
	"strings"
)

const rows = 10

func main() {
	row := []int{1}

	for length := 1; length <= rows; length++ {
		fields := make([]string, len(row))
		for i, value := range row {
			fields[i] = strconv.Itoa(value)
		}
		fmt.Println(strings.Join(fields, " "))

		row = append(row, 0)
		for i := len(row) - 1; i > 0; i-- {
			row[i] += row[i-1]
		}
	}
}
